script/foundry/utils/DeployHelper.sol

```diff
@@ -0,0 +1,369 @@
+        address erc20_,
+        uint256 arbitrationPrice_,
+        uint256 maxRoyaltyApproval_
+    ) JsonDeploymentHandler("main") {
+        erc6551Registry = ERC6551Registry(erc6551Registry_);
+        erc20 = ERC20(erc20_);
+        ARBITRATION_PRICE = arbitrationPrice_;
+        MAX_ROYALTY_APPROVAL = maxRoyaltyApproval_;
+    }
+
+    /// @dev To use, run the following command (e.g. for Sepolia):
+    /// forge script script/foundry/deployment/Main.s.sol:Main --rpc-url $RPC_URL --broadcast --verify -vvvv
+
+    function run(
+        address runDeployer,
+        bool configByMultisig,
+        bool runStorageLayoutCheck,
+        bool writeDeploys_
+    ) public virtual {
+        writeDeploys = writeDeploys_;
+
+        // This will run OZ storage layout check for all contracts. Requires --ffi flag.
+        if (runStorageLayoutCheck) super.run();
+
+        if (block.chainid == 31337) deployer = runDeployer; // set for local before _beginBroadcast
+        _beginBroadcast(); // BroadcastManager.s.sol
+
+        _deployProtocolContracts(runDeployer);
+        if (!configByMultisig) {
+            _configureDeployment();
+        }
+
+        if (writeDeploys) _writeDeployment();
+        _endBroadcast(); // BroadcastManager.s.sol
+    }
+
+    function _deployProtocolContracts(address runDeployer) private {
+        require(address(erc20) != address(0), "Deploy: Asset Not Set");
+
+        string memory contractKey;
+
+        // Core Protocol Contracts
+
+        contractKey = "Governance";
+        _predeploy(contractKey);
+        governance = new Governance(runDeployer);
+        _postdeploy(contractKey, address(governance));
+
+        contractKey = "AccessController";
+        _predeploy(contractKey);
+
+        address impl = address(new AccessController());
+        accessController = AccessController(
+            TestProxyHelper.deployUUPSProxy(impl, abi.encodeCall(AccessController.initialize, address(governance)))
+        );
+        impl = address(0); // Make sure we don't deploy wrong impl
+        _postdeploy(contractKey, address(accessController));
+
+        contractKey = "IPAccountImpl";
+        _predeploy(contractKey);
+        ipAccountImpl = new IPAccountImpl(address(accessController));
+        _postdeploy(contractKey, address(ipAccountImpl));
+
+        contractKey = "ModuleRegistry";
+        _predeploy(contractKey);
+        impl = address(new ModuleRegistry());
+        moduleRegistry = ModuleRegistry(
+            TestProxyHelper.deployUUPSProxy(impl, abi.encodeCall(ModuleRegistry.initialize, address(governance)))
+        );
+        impl = address(0); // Make sure we don't deploy wrong impl
+        _postdeploy(contractKey, address(moduleRegistry));
+
+        contractKey = "IPAccountRegistry";
+        _predeploy(contractKey);
+        ipAccountRegistry = new IPAccountRegistry(address(erc6551Registry), address(ipAccountImpl));
+        _postdeploy(contractKey, address(ipAccountRegistry));
```

Comment:
Do not need to deploy IPAccountRegistry.

Reply:
Resolved with 268d9c3 (pass in casted ipAssetRegistry)